pkg/compactor/job_wait_period.go

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package compactor
+
+import (
+	"context"
+	"path"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/thanos-io/objstore"
+
+	"github.com/grafana/mimir/pkg/storage/tsdb/block"
+)
+
+// jobWaitPeriodElapsed returns whether the 1st level compaction wait period has
+// elapsed for the input job.
+func jobWaitPeriodElapsed(ctx context.Context, job *Job, waitPeriod time.Duration, userBucket objstore.Bucket) (bool, error) {
+	if waitPeriod <= 0 {
+		return true, nil
+	}
+
+	if job.MinCompactionLevel() > 1 {
+		return true, nil
+	}
+
+	if ok, err := jobContainsBlocksUploadedAfter(ctx, job, userBucket, time.Now().Add(-waitPeriod)); err != nil {
```

Comment:
nit: Maybe it makes sense to pass `time.Now()` as an argument to avoid calling it in a loop?

Reply:
I honestly don't think will make any visible difference. We're not calling it millions of times.